lib/Transforms/VectorDistribution.cpp

```diff
@@ -20,6 +20,205 @@ using namespace mlir;
 using namespace mlir::linalg;
 
 namespace {
+
+/// Modified version of the upstream pattern handling more cases.
+/// TODO: Replace upstream once it is ready.
+struct TransferReadExtractPattern
```

Comment:
some docs please

Reply:
(here and below)

Reply:
added docs for the different patterns plus extra comments.